File: src/main/java/com/kerimsenturk/labreport/dto/DiseaseDto.java
package com.kerimsenturk.labreport.dto;

import com.kerimsenturk.labreport.model.enums.DiseaseState;
import com.kerimsenturk.labreport.model.enums.LabRequestType;

import java.time.LocalDateTime;

public class DiseaseDto {
    int id;
    UserDto patient;
    UserDto labTechnician;
    UserDto doctor;
    LabRequestType labRequestType;
    ReportDto pathologicReport;
    ReportDto diagnosticReport;
    DiseaseState diseaseState;
    LocalDateTime creationDate = LocalDateTime.now();

    public DiseaseDto() {
    }

    public DiseaseDto(int id, UserDto patient, UserDto labTechnician, UserDto doctor,
                      LabRequestType labRequestType, ReportDto pathologicReport,
                      ReportDto diagnosticReport, DiseaseState diseaseState, LocalDateTime creationDate) {
        this.id = id;
        this.patient = patient;
        this.labTechnician = labTechnician;
        this.doctor = doctor;
        this.labRequestType = labRequestType;
        this.pathologicReport = pathologicReport;
        this.diagnosticReport = diagnosticReport;
        this.diseaseState = diseaseState;
        this.creationDate = creationDate;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public UserDto getPatient() {
        return patient;
    }

    public void setPatient(UserDto patient) {
        this.patient = patient;
    }

    public UserDto getLabTechnician() {
        return labTechnician;
    }

    public void setLabTechnician(UserDto labTechnician) {
        this.labTechnician = labTechnician;
    }

    public UserDto getDoctor() {
        return doctor;
    }

    public void setDoctor(UserDto doctor) {
        this.doctor = doctor;
    }

    public LabRequestType getLabRequestType() {
        return labRequestType;
    }

    public void setLabRequestType(LabRequestType labRequestType) {
        this.labRequestType = labRequestType;
    }

    public ReportDto getPathologicReport() {
        return pathologicReport;
    }

    public void setPathologicReport(ReportDto pathologicReport) {
        this.pathologicReport = pathologicReport;
    }

    public ReportDto getDiagnosticReport() {
        return diagnosticReport;
    }

    public void setDiagnosticReport(ReportDto diagnosticReport) {
        this.diagnosticReport = diagnosticReport;
    }

    public DiseaseState getDiseaseState() {
        return diseaseState;
    }

    public void setDiseaseState(DiseaseState diseaseState) {
        this.diseaseState = diseaseState;
    }
    public LocalDateTime getCreationDate() {
        return creationDate;
    }

    public void setCreationDate(LocalDateTime creationDate) {
        this.creationDate = creationDate;
    }
}

File: src/main/web-ui/src/domain/dto/DiseaseDto.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DiseaseDto = void 0;
var DiseaseDto = /** @class */ (function () {
    function DiseaseDto(id, patient, labTechnician, doctor, labRequestType, pathologicReport, diagnosticReport, diseaseState, creationDate) {
        this._id = id;
        this._patient = patient;
        this._labTechnician = labTechnician;
        this._doctor = doctor;
        this._labRequestType = labRequestType;
        this._pathologicReport = pathologicReport;
        this._diagnosticReport = diagnosticReport;
        this._diseaseState = diseaseState;
        this._creationDate = creationDate;
    }
    Object.defineProperty(DiseaseDto.prototype, "id", {
        get: function () {
            return this._id;
        },
        set: function (value) {
            this._id = value;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DiseaseDto.prototype, "patient", {
        get: function () {
            return this._patient;
        },
        set: function (value) {
            this._patient = value;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DiseaseDto.prototype, "labTechnician", {
        get: function () {
            return this._labTechnician;
        },
        set: function (value) {
            this._labTechnician = value;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DiseaseDto.prototype, "doctor", {
        get: function () {
            return this._doctor;
        },
        set: function (value) {
            this._doctor = value;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DiseaseDto.prototype, "labRequestType", {
        get: function () {
            return this._labRequestType;
        },
        set: function (value) {
            this._labRequestType = value;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DiseaseDto.prototype, "pathologicReport", {
        get: function () {
            return this._pathologicReport;
        },
        set: function (value) {
            this._pathologicReport = value;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DiseaseDto.prototype, "diagnosticReport", {
        get: function () {
            return this._diagnosticReport;
        },
        set: function (value) {
            this._diagnosticReport = value;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DiseaseDto.prototype, "diseaseState", {
        get: function () {
            return this._diseaseState;
        },
        set: function (value) {
            this._diseaseState = value;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DiseaseDto.prototype, "creationDate", {
        get: function () {
            return this._creationDate;
        },
        set: function (value) {
            this._creationDate = value;
        },
        enumerable: false,
        configurable: true
    });
    return DiseaseDto;
}());
exports.DiseaseDto = DiseaseDto;
